diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3e1f2a7..c4ff9ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -136,6 +136,10 @@ endif( ZLIB_FOUND )
 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DWEBSOCKETPP_STRICT_MASKING")
 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_ASIO_HAS_STD_CHRONO")

+# Set openssl 1.1.0 compatibility for ubuntu 22
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DOPENSSL_API_COMPAT=0x10100000L")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DOPENSSL_API_COMPAT=0x10100000L")
+
 target_include_directories(fc
   PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include
     ${Boost_INCLUDE_DIR}
diff --git a/src/crypto/elliptic_r1.cpp b/src/crypto/elliptic_r1.cpp
index 14b2e60..7f2c9ed 100644
--- a/src/crypto/elliptic_r1.cpp
+++ b/src/crypto/elliptic_r1.cpp
@@ -116,7 +116,7 @@ namespace fc { namespace crypto { namespace r1 {
         if (!BN_bin2bn(msg, msglen, e)) { ret=-1; goto err; }
         if (8*msglen > n) BN_rshift(e, e, 8-(n & 7));
         zero = BN_CTX_get(ctx);
-        if (!BN_zero(zero)) { ret=-1; goto err; }
+        BN_zero(zero);
         if (!BN_mod_sub(e, zero, e, order, ctx)) { ret=-1; goto err; }
         rr = BN_CTX_get(ctx);
         if (!BN_mod_inverse(rr, r, order, ctx)) { ret=-1; goto err; }
