diff --git a/CMakeLists.txt b/CMakeLists.txt
index c20d9b6..b648233 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,6 +68,9 @@ set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Werror")
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-format-zero-length -Wno-implicit-fallthrough")
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat -Wformat-security")

+# Set openssl 1.1.0 compatibility for ubuntu 22
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DOPENSSL_API_COMPAT=0x10100000L")
+
 # fortify can only be used with -O2 and higher, so only enable for release builds
 set (CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -D_FORTIFY_SOURCE=2")
 if (CMAKE_C_COMPILER_ID STREQUAL GNU)
